### PR TITLE
[#683] fix add institution with form

### PR DESCRIPTION
**Pull request by danielkryska**
UI
- [x] Run e2e
- [x] [fix] add/update form
- [x] [fix] set search value (in `/settings`) as it is in URL `instititution` param

Backend
- [x] [fix] Return added/updated institution with response
- [x] [fix] Update should return actualized institution instead of parent

e2e
- [x] Go to an institution should change the value of institution search and results should be closed

### CODE REVIEW

File: s4e-web/src/app/views/settings/profile/profile.component.html [hunk 1]
```diff
@@ -1,33 +1,85 @@
...
+      data-ut="member-institutions-tile"
     >
-      Zmień hasło
-    </button>
-  </footer>
-</section>
+      <img src="./assets/images/ico_institution.svg" alt="Ikona instytucji" width="50" tile-image />
+      <header class="panel__header" tile-title>Instytucje członkowskie</header>
+      <div tile-description>
+        <p *ngFor="let institution of (memberInstitutions$ | async).slice(0, MAX_INSTITUTIONS_TO_DISPLAY_PER_TILE)">
```
Comment (michal-szostak) [inline s4e-web/src/app/views/settings/profile/profile.component.html:61]: This is something that should not be done (calling functions on collections which are iterated in template)
this slice function will be called on every refresh cycle and whole list will be rerendered.
I would suggest adding this slice operation to the pipe of `memberInstitutions$` observable.

File: s4e-web/src/app/views/settings/profile/profile.component.html [hunk 2]
```diff
@@ -1,33 +1,85 @@
...
-      </p>
-    </ng-template>
+    <s4e-tile
+      *ngIf="(administrationInstitutions$ | async).length > 0"
+      data-ut="administrative-institutions-tile"
+    >
+      <img src="./assets/images/ico_institution.svg" alt="Ikona instytucji" width="50" tile-image />
+      <header class="panel__header" tile-title>Instytucje administrowane</header>
+      <div tile-description>
+        <p *ngFor="let institution of (administrationInstitutions$ | async).slice(0, MAX_INSTITUTIONS_TO_DISPLAY_PER_TILE)">
```
Comment (michal-szostak) [inline s4e-web/src/app/views/settings/profile/profile.component.html:31]: The same issue as with `memberInstitutions$` below

**Review by michal-szostak — CHANGES_REQUESTED** [review]
I have only one code issue which should be fixed before merge, a minor thing

**Review by kmarszalek — CHANGES_REQUESTED** [review]
not sure how you resolved this issue 'Update should return actualized institution instead of parent' .
but i dont see how it was an issue in backend in the first place :)
backend looks good.
please add check in InstitutionControllerIntegrationTest that you got right slug.

File: s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerIntegrationTest.java [hunk 3]
```diff
@@ -158,7 +169,15 @@ public void shouldUpdateInstitutionWithEmblem() throws Exception {
                 .contentType(MediaType.APPLICATION_JSON)
                 .content(objectMapper.writeValueAsBytes(request))
                 .with(jwtBearerToken(appUser, objectMapper)))
-                .andExpect(status().isOk());
+                .andExpect(status().isOk())
+                .andExpect(
```
Comment (kmarszalek) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerIntegrationTest.java:173]: .andExpect(jsonPath("$.name",is(equalTo(request.getName()))))

**Review by kmarszalek — APPROVED** [review]

**Pull request merged**